contributors/guide/style-guide.md

```diff
@@ -0,0 +1,683 @@
+It is an extension of the [Kubernetes documentation style-guide].
+
+These are **guidelines**, not rules. Use your best judgement.
+
+- [Cheatsheet](#cheatsheet)
+- [Content design, formatting, and language](#content-formatting-and-language)
+  - [Contact information](#contact-information)
+  - [Dates and times](#dates-and-times)
+  - [Diagrams, images and other assets](#diagrams-images-and-other-assets)
+  - [Document Layout](#document-layout)
+  - [Formatting text](#formatting-text)
+  - [Language, grammar, and tone](#language-grammar-and-tone)
+  - [Moving a document](#moving-a-document)
+  - [Punctuation](#punctuation)
+  - [Quotation](#quotation)
+- [Markdown formatting](#markdown-and-formatting)
+  - [Code Blocks](code-blocks)
+  - [Emphasis](#emphasis)
+  - [Headings](#headings)
+  - [Horizontal Lines](#horizontal-lines)
+  - [Line Length](#line-length)
+  - [Links](#links)
+  - [Lists](#lists)
+  - [Metadata](#metadata)
+  - [Tables](#tables)
+- [Attribution](#attribution)
+
+
+## Cheatsheet
+
+### Cheatsheet: Content design, formatting, and language
+
+**[Contact information:](#contact-information)**
+- Use official Kubernetes contact information.
+
+**[Dates and times:](#dates-and-times)**
+- Format dates as `month day, year`. (December 13, 2018)
+- When conveying a date in numerical form, use [ISO 8601] Format: `yyyy-mm-dd`.
+- Use the 24 hour clock when referencing time.
+- Times for single events (example: KubeCon) should be expressed in an absolute
+  timezone such as Pacific Standard Time (PST) or Coordinated Universal Time
+  (UTC).
+- Times for reoccurring events should be expressed in the PT (Pacific Time)
+  timezone or the most appropriate timezone that adheres to Daylight Savings
+  Time (DST).
+- Supply a link to a globally available time zone converter service.
+  - `http://www.thetimezoneconverter.com/?t=<TIME REFERENCE>&tz=PT%20%28Pacific%20Time%29`
+
+**[Diagrams, images and other assets:](#diagrams-images-and-other-assets)**
+- Images and other assets should be stored in the same directory as the document
+  that is referencing it.
+- Filenames should be lowercase and descriptive of what they are referencing.
+- Avoid excessively large images or include a smaller one while linking to a
+  higher resolution version of the same image.
+- Use the [Kubernetes icon set] for architectural diagrams.
+
+**[Document Layout:](#document-layout)**
+- Documents should follow the general template of:
+  - Document metadata (if appropriate).
+  - Title in `H1` (a single `#`).
+  - A brief description or summary of the document.
+  - A table of contents.
+  - The general body of document.
+- Do not repeat content. Instead link back to the canonical source.
+- Large content or topic shifts should be separated with a horizontal rule.
+
+**[Formatting text:](#formatting-text)**
+- API objects:
+  - Use [camel case], or the format commonly used by default for the language.
```

Comment:
API standards are very specific things, and should probably be out of scope for this (the sig-arch api-approvers team is responsible for those). I'd perhaps link out to the API conventions: `/contributors/devel/sig-architecture/api-conventions.md`

Reply:
That doc itself is definitely the canonical source of truth, but can be pretty intimidating. Do you think linking to the [naming conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#naming-conventions) section directly would  be a good middle ground? It's sort of a "TL;DR" and the focus of this doc is more on the text formatting itself.

Reply:
Updated to point towards the api-conventions doc.